pkg/cli/sql_util.go

```diff
@@ -480,27 +480,53 @@ func makeSQLConn(url string) *sqlConn {
 	}
 }
 
+// getPasswordAndMakeSystemSQLClient prompts for a password if running in secure mode
+// and no certificates have been supplied.
+// The client connection is connected to the `system` database.
+// Attempting to use security.RootUser without valid certificates will return an error.
+func getPasswordAndMakeSystemSQLClient(appName string) (*sqlConn, error) {
+	return makeSQLClient(appName, true /*forceSystemDB*/)
+}
+
 // getPasswordAndMakeSQLClient prompts for a password if running in secure mode
 // and no certificates have been supplied.
 // Attempting to use security.RootUser without valid certificates will return an error.
 func getPasswordAndMakeSQLClient(appName string) (*sqlConn, error) {
-	return makeSQLClient(appName)
+	return makeSQLClient(appName, false /*forceSystemDB*/)
 }
 
 var sqlConnTimeout = envutil.EnvOrDefaultString("COCKROACH_CONNECT_TIMEOUT", "5")
 
-// makeSQLClient connects to the database using the connection
+// makeSystemSQLClient connects to the database using the connection
 // settings set by the command-line flags.
 //
+// If forceSystemDB is set, it also connects it to the `system`
+// database. The --database flag or database part in the URL is then
+// ignored.
+//
 // The appName given as argument is added to the URL even if --url is
 // specified, but only if the URL didn't already specify
 // application_name. It is prefixed with '$ ' to mark it as internal.
-func makeSQLClient(appName string) (*sqlConn, error) {
+func makeSQLClient(appName string, forceSystemDB bool) (*sqlConn, error) {
```

Comment:
Optional suggestion: instead of using the frowned upon pattern of a boolean argument, may I suggest either a typed bool with consts, or otherwise just a string that passes the forced DB, which would allow for `getPasswordAndMakeSystemSQLClient` to specify the expected database itself? That may make this function clearer.